pkg/controllers/dynakube/activegate/internal/statefulset/reconciler.go

```diff
@@ -117,6 +124,19 @@ func (r *Reconciler) calculateActiveGateConfigurationHash(ctx context.Context) (
 	return strconv.FormatUint(uint64(hash.Sum32()), 10), nil
 }
 
+func (r *Reconciler) calculateActiveGateTokenHash(ctx context.Context) (string, error) {
+	tenantToken, err := secret.GetDataFromSecretName(ctx, r.apiReader, types.NamespacedName{
+		Name:      r.dk.ActiveGate().GetTenantSecretName(),
+		Namespace: r.dk.Namespace,
+	}, connectioninfo.TenantTokenKey, log)
+
+	if err != nil {
+		log.Error(err, "secret for activegate token was not available at StatefulSet build time", "dynakube", r.dk.Name)
+	}
```

Comment:
IMHO we should return err and do not change the template if K8S API error encountered. We don't know if the secret/tenant has changed or not.